src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,29 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import cn from 'classnames';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo | null;
+  onHideTodo: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, onHideTodo }) => {
+  const [selectedTodoUser, setSelectedTodoUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    if (selectedTodo) {
+      getUser(selectedTodo.userId).then((user) => setSelectedTodoUser(user));
+    }
+  }, []);
```

Comment:
The useEffect dependency array should include 'selectedTodo' to update 'selectedTodoUser' when 'selectedTodo' changes.